### PR TITLE
Check original title during movie search

### PR DESCRIPTION
**Pull request by olympia**
## Description
This change enable to user to search for a movie based on its original Title using the search function within video library. (Currently the search function is only looking at the title of the movie and not the original title)

## Motivation and Context
Currently the search function is only looking at the title of the movie and not the original title. In case the user only knows the original title and not the title of the movie, he/she will not found the movie. 

## How Has This Been Tested?
There has been a test build built by Jenkins back at the time I tested this. Can't find it any longer to refer.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [X] I have added tests to cover my change
- [X] All new and existing tests passed

### CODE REVIEW

**Review by fuzzard — APPROVED** [review]
Looks reasonable

**Comment by olympia** [comment]
@fuzzard thanks for supporting/ approving. Could you please let me know what is the next step? Do we need a second approval? Who will do the final merge?

**Comment by fuzzard** [comment]
Maybe @DaveTBlake or @Montellese can comment on any potential sql pitfalls

Probably the only concern is large libraries having increased DB query times, but DB stuff is out of my realm. If dave or montellese dont chime in after a few days, can just merge and see how it goes i guess

**Comment by olympia** [comment]
Actually it was Dave writing the code. I am only the tester and submitter as he preferred not to submit changes to video section by himself. So all credits goes for him 👍

**Comment by fuzzard** [comment]
All good, give it a day or two and see if montellese has any input. If not, happy to merge it.

**Comment by Montellese** [comment]
Code wise this looks ok to me. It's hard to estimate the impact on performance though.

It might be confusing to have an entry in the list with only the title being shown which doesn't match the search string at all. Maybe we should add the original title to the label of the hit if the hit was on the original title? Does anyone know how the list of results is sorted / ordered? Is it sorted by hit accuracy or alphabetically? If it is sorted by hit accuracy hits on the original title will probably always end up at the end of the list even if the original title would be a perfect hit.

**Comment by Montellese** [comment]
OK my last concern is irrelevant because the search results are sorted alphabetically in [`CGUIWindowVideoBase::AppendAndClearSearchItems()`](https://github.com/xbmc/xbmc/blob/b17aaaae466ddc74156c5bac61ca5cff62bff5bb/xbmc/video/windows/GUIWindowVideoBase.cpp#L1578).

**Comment by olympia** [comment]
@Montellese I completely agree with your comment; to complete the feature, original title should be added to the search results. However, I am unable to add this and I am not sure I could get anyone to do so. Nevertheless, IMO this would be already an improvement and I think missing the above should not be a blocker. I think returning the title(s) will already ring the bell for most people which movie they are looking for.

**Comment by DaveTBlake** [comment]
The SQL is from me, encouraging @olympia to have an initial  look at core and experience the PR process for the main repo.  There are no bad consequnces on speed of search as far as I can tell. 

Original title is not currently shown in the search results, but I still think it a useful move forwards to include these movies the results.

@olympia nice to see you submit this. The review process of PRs on this repo can be slow, much slower than the scraper repo, this is proceeding well compared to many things. Really it should have someone other than you are me to test it. Anyone got some movies with different titles to the original in their collection?

**Comment by DaveTBlake** [comment]
> There has been a test build built by Jenkins back at the time I tested this. Can't find it any longer to refer.

I'll kick off a Win64 build

**Comment by Montellese** [comment]
> Nevertheless, IMO this would be already an improvement and I think missing the above should not be a blocker. I think returning the title(s) will already ring the bell for most people which movie they are looking for.

It wasn't meant as a rejection just as additional input which is probably worth pursuing in this or a follow-up PR.

**Comment by KarellenX** [comment]
I tested the Original Titles build today and came up with the following...

### Movies

I tested about 20 movies with different names, namely foreign names. Norwegian, Swedish, Japanese, Italian, Chinese, Vietnamese. Interesting that there is a problem with some Asian names. They could not be found. eg

Could **not** be found
```
    <title>Blade Runner: Black Out 2022</title>
    <originaltitle>ブレードランナー ブラックアウト 2022</originaltitle>
```

Could **not** be found
```
    <title>Crouching Tiger, Hidden Dragon</title>
    <originaltitle>卧虎藏龙</originaltitle>
```

Could **not** be found
```
    <title>Kung Fu Hustle</title>
    <originaltitle>功夫</originaltitle>
```

Could be found
```
    <title>Fearless</title>
    <originaltitle>霍元甲</originaltitle>
```

Could be found
```
    <title>The Villainess</title>
    <originaltitle>악녀</originaltitle>
```
Any idea why some are found and others not? Italian, Swedish, English and other Latin type alphabet had no problems.

### TV Shows

Does not work. Not sure if this PR covers TV Shows, but the first show that comes to mind is La Casa De Papel aka Money Heist.


### Episodes

Does not work (which I expected)

Just to confirm, I am using Search from the left sideblade menu, not the Search that uses the Global Search add-on.

original_search

**Comment by Montellese** [comment]
Only movie original titles are supported with this PR. Extending it for tvshows shouldn't be that hard. I'm not sure if Kodi even tracks the original title of episodes.

**Comment by KarellenX** [comment]
>I'm not sure if Kodi even tracks the original title of episodes.

c14 of the episodes table is for original title. 

I have a few episodes with an original title entry, but these are from movies (and the movie.nfo) that I converted into episodes and the nfo file still had the <originaltitle> tag, which indicates that Kodi does read originaltitle for episodes.

But now that I look at TMDB, there is no option to add an original title for episodes at their site.

**Pull request merged**
